File: Exercism Codes/C#/RotationalCipher.cs
using System;
using System.Linq;
public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey) => string.Join("", 
        text.Select(x => 
            (x >= 'a' && x <= 'z') ? 
            (char)(((x - 'a') + shiftKey) % 26 + 'a') : 
            (x >= 'A' && x <= 'Z') ? 
            (char)(((x - 'A') + shiftKey) % 26 + 'A') : 
            x));
}

File: Exercism Codes/PHP/RotationalCipher.php
<?php

declare(strict_types=1);

class RotationalCipher
{
    public function rotate(string $text, int $shift): string
    {
        return implode("", array_map(fn($x) => $this->transform_letter($x, $shift), str_split($text)));
    }

    private function transform_letter(string $x, int $key): string
    {
        return preg_match("/[A-Za-z]/", $x) ? mb_chr($this->is_upper($x) + (((mb_ord($x) - $this->is_upper($x)) + $key) % 26)) : $x;
    }

    private function is_upper(string $x): int
    {
        return preg_match("/[A-Z]/", $x) ? 65 : 97;
    }
}
